engine/src/main/java/nl/inl/blacklab/searches/SearchCacheEntryFromFuture.java

```diff
@@ -20,9 +21,13 @@
 
     final Future<R> future;
 
+    /** A peek at the future result, or null if not available. */
+    private R peekValue;
```

Comment:
I think this can be made immutable with a `final` keyword

Reply:
Agreed!